File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Arrays.sort(participant);
        Arrays.sort(completion);
        // System.out.println(Arrays.toString(participant));
        // System.out.println(Arrays.toString(completion));
        int temp = 0;
        for (int i = 0; i < completion.length; i++){
            // System.out.println(completion[i]);
            // System.out.println(participant[i]);
            if (completion[i].equals(participant[temp])){
                temp +=1;
            }
            else{
                break;
            }
        }
        answer = participant[temp];
        return answer;
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    participant.sort()
    completion.sort()
    if len(completion) == 0:
        return participant[0]
    if participant[-1] != completion[-1]:
        return participant[-1]
    else:
        for i,name in enumerate(participant):
            if name != completion[i]:
                return name
        
    answer = participant[0]
    return answer
    
        
            